ledger-service/http-json/src/main/scala/com/digitalasset/http/json/DomainJsonDecoder.scala

```diff
@@ -25,12 +24,14 @@ class DomainJsonDecoder(
     jsValueToApiValue: (domain.LfType, JsValue) => JsonError \/ lav1.value.Value,
     jsValueToLfValue: (domain.LfType, JsValue) => JsonError \/ domain.LfValue) {
 
+  import com.digitalasset.http.util.ErrorOps._
+
   def decodeR[F[_]](a: String)(
       implicit ev1: JsonReader[F[JsObject]],
       ev2: Traverse[F],
       ev3: domain.HasTemplateId[F]): JsonError \/ F[lav1.value.Record] =
     for {
-      b <- SprayJson.parse(a).leftMap(e => JsonError(e.shows))
+      b <- SprayJson.parse(a).liftErr(JsonError.apply)
```

Comment:
Can't figure out why in this case I need to specify `.apply` for `JsonError` but works fine for `ServerError` without it

Reply:
Because of scala/bug#3664; you have a handwritten companion for `JsonError` but not `ServerError`. You can use the workaround mentioned in https://github.com/scala/bug/issues/3664#issuecomment-320528047; indeed we use this workaround in several places in the daml codebase.

Reply:
We even include the workaround in codegen output:

https://github.com/digital-asset/daml/blob/6685becccc5f04aa6c7833aa1f664dae95ca081f/language-support/scala/codegen/src/main/scala/com/digitalasset/codegen/lf/DamlDataTypeGen.scala#L347-L348